src/components/Articles/Articles.jsx

```diff
@@ -1,27 +1,56 @@
 import { useState, useEffect } from "react";
 import { getArticles } from "../../utils/api";
-import './Articles.css'
+import { Link } from "react-router-dom";
+import Card from "react-bootstrap/Card";
+import CardGroup from "react-bootstrap/CardGroup";
+import "./Articles.css";
 
 export default function ViewArticles() {
   const [articles, setArticles] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
-    getArticles().then((data) => {
-      setArticles(data);
-    });
-  });
+    getArticles()
+      .then((data) => {
+        setArticles(data);
+      })
+      .catch((err) => {
+        setError(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  if (loading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return <p>Error: {error.message}</p>;
+  }
```

Comment:
Good work refactoring this to include isLoading and isError patterns into the app.